src/contract.rs

```diff
@@ -185,7 +185,11 @@ pub fn query<S: Storage, A: Api, Q: Querier>(deps: &Extern<S, A, Q>, msg: QueryM
         QueryMsg::ExchangeRate {} => query_exchange_rate(),
         QueryMsg::Allowance { owner, spender, .. } => try_check_allowance(deps, owner, spender),
         QueryMsg::Minters { .. } => query_minters(deps),
-        _ => authenticated_queries(deps, msg),
+
+        // Authenticated queries
+        QueryMsg::Balance { .. } | QueryMsg::TransferHistory { .. } => {
+            authenticated_queries(deps, msg)
+        }
```

Comment:
This change is kind of unnecessary. it'll just get more verbose if we add variants to `QueryMsg`

Reply:
Yes, but it prevents users to get panicked if they misspell a query message